org.eclipse.jdt.ls.filesystem/src/org/eclipse/jdt/ls/core/internal/filesystem/JDTLSFilesystemActivator.java

```diff
@@ -15,17 +24,60 @@
 public class JDTLSFilesystemActivator implements BundleActivator {
 
 	private static BundleContext context;
+	private static final String JAVA_LS_PLUGIN_ID = "org.eclipse.jdt.ls.core";
+	private static final String JAVA_RESOURCE_FILTERS = "java.project.resourceFilters";
+	private static final String JAVA_RESOURCE_FILTERS_DEFAULT = "node_modules::\\.git";
+	private List<Pattern> resourcePatterns; 
+	private String resourceFilters;
+	private static JDTLSFilesystemActivator instance;
 
 	static BundleContext getContext() {
 		return context;
 	}
 
 	public void start(BundleContext bundleContext) throws Exception {
 		JDTLSFilesystemActivator.context = bundleContext;
+		JDTLSFilesystemActivator.instance = this;
+		configureResourceFilters();
+	}
+
+	private void configureResourceFilters() {
+		IEclipsePreferences eclipsePreferences = InstanceScope.INSTANCE.getNode(JAVA_LS_PLUGIN_ID);
+		if (eclipsePreferences != null) {
+			resourceFilters = eclipsePreferences.get(JAVA_RESOURCE_FILTERS, JAVA_RESOURCE_FILTERS_DEFAULT);
+			eclipsePreferences.addPreferenceChangeListener(new IPreferenceChangeListener() {
+
+				@Override
+				public void preferenceChange(PreferenceChangeEvent event) {
+					if (event.getNewValue() instanceof String newValue && Objects.equals(JAVA_RESOURCE_FILTERS, event.getKey()) && !Objects.equals(resourceFilters, event.getNewValue())) {
+						resourceFilters = newValue;
+						setResourcePatterns();
+					}
+				}
+			});
+		} else {
+			resourceFilters = JAVA_RESOURCE_FILTERS_DEFAULT;
+		}
+		setResourcePatterns();
+	}
+
+	protected void setResourcePatterns() {
+		resourcePatterns = new ArrayList<>();
+		for (String element: resourceFilters.split("::")) {
```

Comment:
Can you leave a space between `element` and `:`

Reply:
Fixed